athena-jdbc/src/test/java/com/amazonaws/connectors/athena/jdbc/integ/RedshiftIntegTest.java

```diff
@@ -0,0 +1,281 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.jdbc.integ;
+
+import com.amazonaws.athena.connector.integ.data.ConnectorVpcAttributes;
+import com.amazonaws.athena.connector.integ.stacks.ConnectorWithVpcStack;
+import com.amazonaws.athena.connector.integ.IntegrationTestBase;
+import com.amazonaws.services.athena.model.Row;
+import com.google.common.collect.ImmutableMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.Test;
+import software.amazon.awscdk.core.RemovalPolicy;
+import software.amazon.awscdk.core.SecretValue;
+import software.amazon.awscdk.core.Stack;
+import software.amazon.awscdk.services.ec2.SecurityGroup;
+import software.amazon.awscdk.services.ec2.Vpc;
+import software.amazon.awscdk.services.ec2.VpcAttributes;
+import software.amazon.awscdk.services.iam.PolicyDocument;
+import software.amazon.awscdk.services.redshift.Cluster;
+import software.amazon.awscdk.services.redshift.ClusterType;
+import software.amazon.awscdk.services.redshift.Login;
+import software.amazon.awscdk.services.redshift.NodeType;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Integration-tests for the DynamoDB connector using the Integration-test module.
+ */
+public class RedshiftIntegTest extends IntegrationTestBase
+{
+    private static final Logger logger = LoggerFactory.getLogger(RedshiftIntegTest.class);
+
+    private static final String REDSHIFT_DB_NAME = "public";
+    private static final String REDSHIFT_DB_PORT = "5439";
+    private static final String REDSHIFT_DB_USERNAME = "integusername";
+    private static final String REDSHIFT_DB_PASSWORD = "IntegPassword1";
+    private static final String REDSHIFT_TABLE_MOVIES = "movies";
+    private static final String REDSHIFT_TABLE_BDAY = "bday";
+
+    private static final long sleepDelayMillis = 120_000L;
+
+    private final String lambdaFunctionName;
+    private final String clusterName;
+    private final String clusterEndpoint;
+    private final Map<String, String> environmentVars;
+
+    public RedshiftIntegTest()
+    {
+        lambdaFunctionName = getLambdaFunctionName();
+        clusterName = "redshift-" + UUID.randomUUID();
+        clusterEndpoint = clusterName + ".c9afbdb0synx.us-east-1.redshift.amazonaws.com";
```

Comment:
why are we hard coding any portion of the hostname/endpoint? are we saying no one should use this outside of us-east-1?

Reply:
Redshift doesn't give you the ability to customize the endpoint. Furthermore, they don't provide a good API for extracting the endpoint after-the-fact (if they did, we would still have to break the CloudFormation process into 2 separate stacks since the Lambda also needs the connection_string containing the endpoint). I did, however, move this to the config file so it would be easier to change by the user if need be.